src/ClientFactory.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace oliverlorenz\reactphpmqtt;
+
+use oliverlorenz\reactphpmqtt\protocol\Version;
+use React\Dns\Resolver\Factory as DnsResolverFactory;
+use React\EventLoop\Factory as EventLoopFactory;
+
+class ClientFactory
+{
+    public static function createClient(Version $version, $resolverIp)
```

Comment:
I'll just set the default to `8.8.8.8` for now. Then we can see if we can change how it works later.

I would've thought that you'd be able to just use whatever DNS resolver your web browser would usually use, but I don't really know how that works.